engine/resources/resources_test.go

```diff
@@ -286,6 +286,44 @@ func TestResources1(t *testing.T) {
 			cleanup: func() error { return os.Remove(f) },
 		})
 	}
+	{
+		r := makeRes("exec", "x1")
+		res := r.(*ExecRes) // if this panics, the test will panic
+		s := "hello, world"
+		f := "/tmp/whatever"
+		res.Cmd = fmt.Sprintf("echo '%s' > '%s'", s, f)
+		res.Shell = "/bin/bash"
+		res.IfCmd = "! diff <(cat /tmp/whatever) <(echo hello, world)"
+		res.IfShell = "/bin/bash"
+		res.WatchCmd = fmt.Sprintf("/usr/bin/inotifywait -e modify -m %s", f)
+		res.Env = map[string]string{
+			"boiling": "one hundred", "freezing": "zero", "room": "twenty-five",
+			"house": "twenty-two", "canada": "negative thirty",
+		}
+		//res.WatchShell = "/bin/bash"
+
+		timeline := []Step{
```

Comment:
Do we ever check in here somewhere that `Env` actually did anything? IOW if you took all the `Env` code in CheckApply and other methods away, would this test still pass? If so, then it's missing something.

Reply:
I just tested it with a print. It does in fact add the correct variables to env like it is supposed to. However, if I take out necessary code, the test still passes... I'm going to keep working on it.

Reply:
Nowhere in this test are you actually testing the ENV parts :/